Fix DAP_vendor.c

gaborcsapo opened the pull request:
It was synced to DAPLink master instead of develop branch, which had this
minor difference that breaks builds.

yangtechen commented:
For bootloader build, it has an error that says
`../../../source/hic_hal/stm32/stm32h743ii/extended_features/source/DAP_vendor_ex.c:1:10: fatal error: DAP.h: No such file or directory`

yangtechen commented on records/hic_hal/stm32h743ii.yaml:
```diff
@@ -18,14 +18,12 @@ common:
         - source/hic_hal/stm32/stm32h743ii/STM32H7xx_HAL_Driver
         - source/hic_hal/stm32/stm32h743ii/STM32H7xx_HAL_Driver/Inc
         - source/hic_hal/stm32/stm32h743ii/STCubeGenerated/Inc
-        - source/hic_hal/stm32/stm32h743ii/extended_features/include
```
Deletion of this line would make `main_interface.c` fail to include `i2c.h`. My opinion would be keeping this one and remove the line `source/hic_hal/stm32/stm32h743ii/extended_features/include` in `records/board/stm32h743ii_if.yaml`

gaborcsapo replied:
Actually the issue is deeper than that...

yangtechen replied:
I am not quite sure how the build file should be look like. I just proposed a workaround that would make both interface and bootloader build locally.

gaborcsapo committed:
Fix project setup

- move from using the default empty board info to our board specific one
with the prerun_board_init function defined
- keep the extended features separate from hic_hal module in the project
config for clarity
- accidentally synced DAP_vendor.c to the Daplink master in the previous
commit, instead of the develop branch

yangtechen reviewed (approved):
Both interface and bootloader can be built successfully.

eleenest reviewed (approved):
I reviewed the changes, no red flags that I noted.
